src/Microsoft.Restier.Core/Extensions/ApiBaseExtensions.cs

```diff
@@ -54,20 +54,6 @@ public static class ApiBaseExtensions
             return api.ServiceProvider.GetService<T>();
         }
 
-        /// <summary>
```

Comment:
Yeah. This is a bit half-hearted right? The ServiceProvider is accessible though. Question is: Do we really need these methods, or is exposing the ServiceProvider enough. I'll put it back in for now.

Reply:
You're right, you DO have access to the ServiceProvider, so this feels lame. However, making it easier to get the service you need has the added benefit of abstracting away where you get it from... which means if we do something different later, we can keep the method so the end user doesn't need to modify code.

It will be entirely unnecessary in .NET 5 when we can make the RestierController middleware and let the API instance use constructor injection excludively.

Reply:
Yeah, normally abstraction is a reasonable thing. But since it's an anti-pattern already and MS made it a double anti-pattern by having hundreds of libraries add extension methods to IServiceProvider (including themselves e.g. GetRequiredService), the abstraction thing is debatable. Users are probably better off by having an IServiceProvider.

Reply:
Thanks for getting it back in :)